### PR TITLE
STOR-2078: Add feature gate VolumeAttributesClass

### PR DESCRIPTION
**Pull request by dfajmon**

### CODE REVIEW

File: features/features.go [hunk 1]
```diff
@@ -448,6 +448,13 @@ var (
 					enhancementPR("https://github.com/kubernetes/enhancements/issues/3476").
 					enableIn(configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
 					mustRegister()
+	
+	FeatureGateVolumeAttributesClass = newFeatureGate("VolumeAttributesClass").
```
Comment (JoelSpeed) [inline features/features.go:452]: This is currently beta upstream but is not enabled by default, the motivation here then is to enable this in tech preview, but not default clusters.

When it goes default upstream, will the feature go locked to true straight away? Could this not being enabled for us cause an issue with the rebase?

Do we know when it is likely to go GA?

Do you have a link to the relevant upstream/downstream enhancements about this feature?

Comment (dfajmon) [reply features/features.go:452]: > When it goes default upstream, will the feature go locked to true straight away?

Probably, yes. 

> Could this not being enabled for us cause an issue with the rebase?

When it goes enabled upstream, then we will enable it in OCP too. It's likely to be GA at that point.

> Do we know when it is likely to go GA?

Earliest is 1.33.

>Do you have a link to the relevant upstream/downstream enhancements about this feature?

https://github.com/kubernetes/enhancements/issues/3751

**Pull request merged**